distributed/executor.py

```diff
@@ -974,6 +992,45 @@ def cancel(self, futures):
         """
         return sync(self.loop, self._cancel, futures)
 
+    def publish_dataset(self, data, name):
+        """
+        Store a named reference to the data in the scheduler for other executors
+
+        Parameters
+        ----------
+        data: list of futures or dask collection
+        name: str
+            handle by which the stored data should be known
+
+        Returns
+        -------
+        None
+        """
+        keys = [tokey(f.key) for f in futures_of(data)]
+        self._send_to_scheduler({'op': 'publish-data', 'keys': keys,
```

Comment:
Perhaps we should move away from the term `publish-data` to `publish-dataset` to avoid conceptual overlap with `scatter`